ros/src/twist_controller/dbw_node.py

```diff
@@ -63,9 +63,9 @@ def __init__(self):
         self.previous_loop_time = rospy.get_rostime()
         self.previous_debug_time = rospy.get_rostime()
 
-        self.throttle_pid = pid.PID(kp=0.2, ki=0.01, kd=0.1, mn=decel_limit, mx=0.5 * accel_limit)
-        self.brake_pid = pid.PID(kp=30.0, ki=0.001, kd=5.0, mn=brake_deadband, mx=2000)
-        self.steering_pid = pid.PID(kp=1.0, ki=0.001, kd=0.5, mn=-max_steer_angle, mx=max_steer_angle)
+        self.throttle_pid = pid.PID(kp=0.5, ki=0.0, kd=0.0, mn=decel_limit, mx=0.5 * accel_limit)
```

Comment:
I checked your changes, but don't agree they give better results on throttle and braking.
For throttle - the reference velocity is 9 mph, but car only goes to ~8.6mph. The integral factor in throttle PID was there to help combat this gap.
For brake - if I accelerate the car to say 40mph and the turn on drive by wire, car decelerators to a full stop before accelerating to reference velocity of 9mph. That's not a safe or sensible behaviour.
Please rework PID values so that above are fixed.